ui/lib/core/addon/components/masked-input.hbs

```diff
@@ -64,4 +58,32 @@
   >
     <Icon @name={{if this.showValue "eye" "eye-off"}} />
   </button>
-</div>
\ No newline at end of file
+</div>
+
+{{! CONFIRM DOWNLOAD MODAL }}
+<Modal
+  @title="Download secret value?"
+  @onClose={{action (mut this.modalOpen) false}}
+  @isActive={{this.modalOpen}}
+  @type="warning"
+>
+  <section class="modal-card-body">
+    This download is
+    <strong>unencrypted</strong>. Are you sure you want to download this secret data as plaintext?
+  </section>
+  <footer class="modal-card-foot modal-card-foot-outlined">
+    <DownloadButton
+      class="button is-primary"
+      @filename={{or @name "secret-value"}}
+      @data={{@value}}
+      @stringify={{true}}
+      aria-label="Download secret value"
+      {{on "click" (fn (mut this.modalOpen) false)}}
```

Comment:
@zofskeez - There's a click event in the `<DownloadButton>` that handles the eventual download. It works fine locally but do you foresee any issue with stacking events like this?

Reply:
I think it would be better to pass the action up from the `handleDownload` method in `DownloadButton`. Something like:
```js
async handleDownload() {
  try {
    // download was a success
    if (this.args.onSuccess) {
      this.args.onSuccess();
    }
  } catch {}
}
```

Reply:
Yes! That's what I was thinking 🤔 Will refactor!